### ISSUE TITLE
`jsx-a11y/label-has-for` is deprecated in favor of `jsx-a11y/label-has-associated-control`.

### ISSUE THREAD

**Issue by qqii** [Shopify/web-configs#484]
## Overview

> # jsx-a11y/label-has-for
>
> ❌ This rule is deprecated. It was replaced by [`jsx-a11y/label-has-associated-control`](https://github.com/infofarmer/eslint-plugin-jsx-a11y/blob/main/docs/rules/label-has-associated-control.md).
>
> 🚫 This rule is _disabled_ in the following configs: ☑️ `recommended`, 🔒 `strict`.

https://github.com/infofarmer/eslint-plugin-jsx-a11y/blob/main/docs/rules/label-has-for.md

Rule `jsx-a11y/label-has-for` should be removed.

## Motivation

The rule is deprecated, and we [already enable the replacement](https://github.com/Shopify/web-configs/blob/6c51fc60ef04a6229c5498acae988b49bedb4a18/packages/eslint-plugin/lib/config/react.js#L269-L272). 

## Labels

- [ ] Add the `Type: Enhancement` label to this issue.

### Area

- [ ] Add any relevant `Area: eslint-plugin` labels to this issue

### Scope

- Is this issue related to a specific package?

  - [ ] Tag it with the `Package: @shopify/eslint-plugin` label.

---

## Checklist

- [ ] Please delete the labels section before submitting your issue
- [ ] I have described this issue in a way that is actionable (if possible)

**Comment by qqii** [comment]
I don't seem to be able to add labels.
